fv3core/fv3core/initialization/baroclinic.py

```diff
@@ -0,0 +1,522 @@
+):
+    ps[:] = jablo_init.surface_pressure
+    delp[:, :, :-1] = initialize_delp(ps, ak, bk)
+    pe[:] = initialize_edge_pressure(delp, ptop)
+    peln[:] = initialize_log_pressure_interfaces(pe, ptop)
+    pk[:], pkz[:] = initialize_kappa_pressures(pe, peln, ptop)
+    eta[:-1], eta_v[:-1] = jablo_init.compute_eta(ak, bk)
+
+
+def baroclinic_initialization(
+    eta,
+    eta_v,
+    peln,
+    qvapor,
+    delp,
+    u,
+    v,
+    pt,
+    phis,
+    delz,
+    w,
+    lon,
+    lat,
+    lon_agrid,
+    lat_agrid,
+    ee1,
+    ee2,
+    es1,
+    ew2,
+    ptop,
+    adiabatic,
+    hydrostatic,
+    nx,
+    ny,
+):
+    """
+    Calls methods that compute initial state via the Jablonowski perturbation test case
+    Transforms results to the cubed sphere grid
+    Creates an initial baroclinic state for u(x-wind), v(y-wind), pt(temperature),
+    phis(surface geopotential)w (vertical windspeed) and delz (vertical coordinate layer
+    width)
+
+    Inputs lon, lat, lon_agrid, lat_agrid, ee1, ee2, es1, ew2, ptop are defined by the
+           grid and can be computed using an instance of the MetricTerms class.
+    Inputs eta and eta_v are vertical coordinate columns derived from the ak and bk
+           variables, also found in the Metric Terms class.
+    """
+
+    # Equation (2) for v
+    # Although meridional wind is 0 in this scheme
+    # on the cubed sphere grid, v is not 0 on every tile
+    initialize_zonal_wind(
+        v,
+        eta,
+        eta_v,
+        lon,
+        lat,
+        east_grid_vector_component=ee2,
+        center_grid_vector_component=ew2,
+        islice=slice(0, nx + 1),
+        islice_grid=slice(0, nx + 1),
+        jslice=slice(0, ny),
+        jslice_grid=slice(1, ny + 1),
+        axis=1,
+    )
+
+    initialize_zonal_wind(
+        u,
+        eta,
+        eta_v,
+        lon,
+        lat,
+        east_grid_vector_component=ee1,
+        center_grid_vector_component=es1,
+        islice=slice(0, nx),
+        islice_grid=slice(1, nx + 1),
+        jslice=slice(0, ny + 1),
+        jslice_grid=slice(0, ny + 1),
+        axis=0,
+    )
+
+    # Compute cell lats in the midpoint of each cell edge
+    lat2, lat3, lat4, lat5 = compute_grid_edge_midpoint_latitude_components(lon, lat)
+    slice_3d = (slice(0, nx), slice(0, ny), slice(None))
+    slice_2d = (slice(0, nx), slice(0, ny))
+    # initialize temperature
+    t_mean = jablo_init.horizontally_averaged_temperature(eta)
+    pt[slice_3d] = cell_average_nine_components(
+        jablo_init.temperature,
+        [eta, eta_v, t_mean],
+        lat,
+        lat_agrid,
+        lat2,
+        lat3,
+        lat4,
+        lat5,
+        slice_2d,
+    )
+
+    # initialize surface geopotential
+    phis[slice_2d] = cell_average_nine_components(
+        jablo_init.surface_geopotential_perturbation,
+        [],
+        lat,
+        lat_agrid,
+        lat2,
+        lat3,
+        lat4,
+        lat5,
+        slice_2d,
+    )
```

Comment:
`cell_average_nine_components` and `compute_grid_edge_midpoint_latitude_components` are doing something low level and pretty complex - I can't tell what they are doing. compute_grid_edge_midpoint_latitude_components is also only used for this second function.

Is there any change you can make to have it be more clear what's happening at this function's level, by pushing the low-level logic down one call level? For example, even if it's less efficient, by pushing the call to compute_grid_edge_midpoint_latitude_components down one level (removing many low-level call arguments from this function's level), and by pulling the slicing operation (which is a higher level operation you're already using here in the assignment slice) up to this level (e.g. by passing `lat[slice_2d]` and `lat_agrid[slice_2d]`)?